streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListenerBus.scala

```diff
@@ -17,17 +17,28 @@
 
 package org.apache.spark.streaming.scheduler
 
-import java.util.concurrent.atomic.AtomicBoolean
+import org.apache.spark.scheduler.{LiveListenerBus, SparkListener, SparkListenerEvent}
+import org.apache.spark.util.ListenerBus
 
-import org.apache.spark.Logging
-import org.apache.spark.util.AsynchronousListenerBus
+/**
+ * A Streaming listener bus to forward events in WrappedStreamingListenerEvent to StreamingListeners
+ */
+private[streaming] class StreamingListenerBus(sparkListenerBus: LiveListenerBus)
+  extends SparkListener with ListenerBus[StreamingListener, StreamingListenerEvent] {
 
-/** Asynchronously passes StreamingListenerEvents to registered StreamingListeners. */
-private[spark] class StreamingListenerBus
-  extends AsynchronousListenerBus[StreamingListener, StreamingListenerEvent]("StreamingListenerBus")
-  with Logging {
+  sparkListenerBus.addListener(this)    // for getting callbacks on spark events
 
-  private val logDroppedEvent = new AtomicBoolean(false)
+  def post(event: StreamingListenerEvent) {
```

Comment:
Added `stop` and test. I also added `start` method because `StreamingContext.stop` will just return if `StreamingContext.start` is not called.

Right now, the behavior is registering StreamingListenerBus to the Spark listener bus in `StreamingContext.start` and unregistering it in `StreamingContext.stop`.